### ISSUE TITLE
Errors upgrading from 1.7.6.9 to 1.7.7.0 - League\Tactician\Bundle\TacticianBundle not found

### ISSUE THREAD

**Issue by RayKoni** [PrestaShop/PrestaShop#22288]
Upgrading from 1.7.6.9 to 1.7.7.0
Gives following error messages

Errors:

[INTERNAL] /file_path/app/AppKernel.php line 59 - Error: Class 'League\Tactician\Bundle\TacticianBundle' not found #0 /file_path/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(488): AppKernel->registerBundles() #1 /file_path/vendor/symfony/symfony/src/Symfony/Component/HttpKernel/Kernel.php(132): Symfony\Component\HttpKernel\Kernel->initializeBundles() #2 /file_path/modules/autoupgrade/classes/UpgradeTools/SymfonyAdapter.php(74): Symfony\Component\HttpKernel\Kernel->boot() #3 /file_path/modules/autoupgrade/classes/UpgradeTools/CoreUpgrader/CoreUpgrader17.php(50): PrestaShop\Module\AutoUpgrade\UpgradeTools\SymfonyAdapter->initAppKernel() #4 /file_path/modules/autoupgrade/classes/UpgradeTools/CoreUpgrader/CoreUpgrader.php(85): PrestaShop\Module\AutoUpgrade\UpgradeTools\CoreUpgrader\CoreUpgrader17->initConstants() #5 /file_path/modules/autoupgrade/classes/TaskRunner/Upgrade/UpgradeDb.php(41): PrestaShop\Module\AutoUpgrade\UpgradeTools\CoreUpgrader\CoreUpgrader->doUpgrade() #6 /file_path/oshu_d3sk/autoupgrade/ajax-upgradetab.php(53): PrestaShop\Module\AutoUpgrade\TaskRunner\Upgrade\UpgradeDb->run() #7 {main}

It appears that files are not copied to where they are supposed to.
Manually or through 1-click upgrade gives same results.

Server info
PS-version 1.7.6.9
PHP-versie: 7.2.34
MySQL versie: 10.1.48-MariaDB
MySQL-engine: InnoDB
MySQL-driver: DbPDO

Hope anyone can shed some light on this.

TIA.

**Comment by matks** [comment]
@RayKoni Do you have files inside folder `vendor/league/` ? This is where TacticianBundle should be.

Indeed it seems to indicate that `It appears that files are not copied to where they are supposed to.` as you mention. Can you check the user you use for autoupgrade has the access rights that allow it to write files in the full prestashop directory?

**Comment by RayKoni** [comment]
Ist of all thnx for trying to hrlp me out.

User has all the right permissions.
When I manually insert TacticianBundle.php(which I find in the prestashop/vendorleague/tactician-bundle/src dir) it continues and it stops at the EventManager.php file.
I have also removed non PS themes and modules but still same effect.
Next try is a clean install.

**Comment by matks** [comment]
> Ist of all thnx for trying to hrlp me out.
> 
> User has all the right permissions.
> When I manually insert TacticianBundle.php(which I find in the prestashop/vendorleague/tactician-bundle/src dir) it continues and it stops at the EventManager.php file.

Don't try to add manually all files, there are hundred 😅 

Moreover TacticianBundle.php was already installed on PS 1.7.6 so it should not be different between PS 1.7.6 and PS 1.7.7 . I think something wrong happened during you upgrade. It should be a simple "unzip the file" step. It's not supposed to "miss" some files.

**Comment by RayKoni** [comment]
Wasn't planning to 😅
Yes but the whole league dir is deleted and then recreated.
TacticianBundle.php is in the src dir in the prestashop.zip unpacking the zip, placing TacticianBundle.php in the right dir and then creating a new zip solved that but there seems to be more.

**Comment by hibatallahAouadni** [comment]
Hello @RayKoni 

I didn't manage to reprocude your issue while upgrading PS1.7.6.9 to PS1.7.7.0, the only problem I encounter is #22250
@matks thanks for your quick replies :rocket: 
@RayKoni could you please edit your issue and add clear steps to reproduce the issue, cause it seems you made some changes before upgrading.

Thanks!

**Comment by RayKoni** [comment]
Hi @hibatallahAouadni,
I did nothing else then firing up the 1-click upgrade 
Backup goes ok.
Deleting goes ok.
Copying goes ok.(So it says)
The last lines are:

[INTERNAL] /file_path/modules/autoupgrade/vendor/composer/ClassLoader.php line 444 - include(/file_path/vendor/composer/../league/tactician-bundle/TacticianBundle.php): failed to open stream: No such file or directory
[INTERNAL] /file_path/modules/autoupgrade/vendor/composer/ClassLoader.php line 444 - include(): Failed opening '/file_path/vendor/composer/../league/tactician-bundle/TacticianBundle.php' for inclusion (include_path='/file_path/vendor/pear/pear_exception:/file_path/vendor/pear/console_getopt:/file_path/vendor/pear/pear-core-minimal/src:/file_path/vendor/pear/archive_tar:.:/opt/../share/pear')

And then I get the error

The only changes I made was after the initial error.
I have attached a txt file with Version comparison lists
Comparison of the **These files will be deleted** list and the **These files will be replaced** shows that neither /vendor/league/tactician-bundle/TacticianBundle.php nor  /vendor/doctrine/common/lib/Doctrine/Common/EventManager.php are in the to be replaced list yet they are in the to be deleted list.
[Version Comparison.txt](https://github.com/PrestaShop/PrestaShop/files/5660692/Version.Comparison.txt)

**Comment by RayKoni** [comment]
Problem solved.
I have activated the english language and made it default.
That did the trick.

**Comment by hibatallahAouadni** [comment]
Glad to hear :tada: 
Thanks @RayKoni for your feedback!

**Comment by gnojus** [comment]
Hello!
May I revive this?
I recently had the same issue while upgrading prestashop `1.7.6.8` to `1.7.7.0`, with the errors saying exactly the same.
I also noticed a change in vendor directory structure, which seems to have caused the problem. For comparison:
```
$ tree -L 1 prestashop_1.7.6.8/vendor/league/tactician-bundle/
prestashop_1.7.6.8/vendor/league/tactician-bundle/
├── DependencyInjection
├── Handler
├── LICENSE
├── Middleware
├── Resources
└── TacticianBundle.php
```
```
$ tree -L 2 prestashop_1.7.7.0/vendor/league/tactician-bundle/
prestashop_1.7.7.0/vendor/league/tactician-bundle/
└── src
    ├── Command
    ├── DependencyInjection
    ├── Handler
    ├── Middleware
    ├── Resources
    ├── Security
    └── TacticianBundle.php
```
thus the `TacticianBundle.php` file is nested deeper down in later release, and php cannot find it anymore as it is moved to `src` subfolder.
Could someone help to resolve this issue and guide how to upgrade successfully, because as of now I have broken installation. 
Thanks

**Comment by matks** [comment]
Hi @nojus297 you are right.

In PrestaShop 1.7.6, [thephpleague/tactician-bundle](https://github.com/thephpleague/tactician-bundle) v0.4.1 was used.
In PrestaShop 1.7.7, [thephpleague/tactician-bundle](https://github.com/thephpleague/tactician-bundle) v1.2.0 is used.

However this change should be handled by Composer. There might be some composer cache so your shop still looks for old directory.

ping @atomiix FYI

**Comment by gnojus** [comment]
Hi and thanks for the reply.

Update: I tried to rollback using the autoupgrade tool, which failed also but at least reverted the files. After that I reverted the database using manual backup and the shop was working again in `1.7.6.8`.
I then cleared the cache using the button in back office and changed default language of the site to English, as reported earlier by **RayKoni**.

Then I tried upgrading again and this time the upgrade completed successfully, so I guess something fixed it.
